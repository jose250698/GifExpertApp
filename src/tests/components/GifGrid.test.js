import {shallow} from 'enzyme';
import React from 'react';
import '@testing-library/jest-dom';
import GifGrid from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');


describe('<GifGrid /> tests', () => {

    const category = 'EVO X';
    // let wrapper = shallow(<GifGrid category={category} />);

    test('should show component correctly', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
    });

    test('should show items when images are loaded', () => {
        let gifs = [{
            id: 'abc', 
            title: 'PÃ¡gina', 
            url: '/icon.png'
        }];
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
        const wrapper = shallow(<GifGrid category={category} />);
        
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    });

})
