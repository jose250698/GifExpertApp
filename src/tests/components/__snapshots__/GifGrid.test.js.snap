// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifGrid /> tests should show component correctly 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    EVO X
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`<GifGrid /> tests should show items when images are loaded 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    EVO X
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="abc"
      key="abc"
      title="PÃ¡gina"
      url="/icon.png"
    />
  </div>
</Fragment>
`;
